src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py

```diff
@@ -268,7 +270,13 @@ def go_back(self, checked=False):
     def on_breadcrumb_clicked(self, tgt_level):
         if int(tgt_level) + 1 != len(self.channels_stack):
             self.go_back_to_level(tgt_level)
+        elif isinstance(self.model, SearchResultsModel) and len(self.channels_stack) == 1:
+            # In case of remote search, when only the search results are on the stack,
+            # we must keep the txt_filter (which contains the search term) before resetting the view
+            text_filter = self.model.text_filter
+            self.reset_view(text_filter=text_filter)
```

Comment:
**issue:** I could be wrong, but it seems that the base class (ChannelContentsWidget) adopted his behavior for 3 different views (search, discovery, channel content). It is unclear, which parts of the class are used by which views.
For those who are not the author of the class, It is hard to understand the logic and hard to make proper changes to the class.

**suggestion:** use different implementations of ChannelContentsWidget to different views (search, discovery, channel content)

Reply:
> use different implementations of ChannelContentsWidget to different views (search, discovery, channel content)

pre-7.5 versions used this approach. This resulted in *massive* code duplication because Qt does not allow for inheritance in `.ui`-defined classes. So, we were left with two options (besides the copy-paste):
 * stop using QtDesigner for Channels stuff and lose the ability to see and manipulate the layout visually
 * create a single uber-class and disable its features as necessary

The current state of affairs it the result of choosing the second option...

Reply:
Can you give me a link to this "QT feature"?

Reply:
> create a single uber-class and disable its features as necessary

If it is the only possible solution (if we assume that QtDesigner is mandatory), then probably, we should set boundaries inside this class with more accuracy.

Reply:
>we should set boundaries inside this class with more accuracy.

Indeed, we should. ChannelConentsWidget was never *designed*, but instead born as a merge of several classes. I'm not going to refactor it until we plan some big new features for it. It works fine for now.